frontend/packages/console-shared/src/actions/modify-health-checks.ts

```diff
@@ -0,0 +1,52 @@
+import * as _ from 'lodash';
+import { K8sKind, K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import { KebabOption } from '@console/internal/components/utils';
+
+const healthChecksAdded = (resource: K8sResourceKind): boolean => {
+  const containers = resource?.spec?.template?.spec?.containers;
+  return _.every(
+    containers,
+    (container) => container.readinessProbe || container.livenessProbe || container.startupProbe,
+  );
+};
+
+const healthChecksUrl = (model: K8sKind, obj: K8sResourceKind): string => {
+  const {
+    kind,
+    metadata: { name, namespace },
+  } = obj;
+  const resourceKind = model.crd ? referenceFor(obj) : kind;
+  const containers = obj?.spec?.template?.spec?.containers;
+  const containerName = containers?.[0]?.name;
+  return `/k8s/ns/${namespace}/${resourceKind}/${name}/containers/${containerName}/health-checks`;
+};
+
+export const AddHealthChecks = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
```

Comment:
All actions should be contributed through extensions and we should be adding health checks to the console app and not shared.

However considering this issue also is requesting a specific order. We cannot correctly satisfy the ask through extensions because extensions are added at a specific point in the list. Furthermore the kebab actions extensions doesn't support ordering.

fyi @vojtechszocs we need to add action ordering along with contributing all actions through extensions.

Reply:
Moved health checks action to console app.